subprojects/signing/src/main/java/org/gradle/plugins/signing/Sign.java

```diff
@@ -218,20 +229,45 @@ public void execute(PublishArtifact artifact) {
             configuration.getAllArtifacts().whenObjectRemoved(new Action<PublishArtifact>() {
                 @Override
                 public void execute(final PublishArtifact publishArtifact) {
-                    signatures.remove(Iterables.find(signatures, new Predicate<Signature>() {
-                        @Override
-                        public boolean apply(Signature input) {
-                            return input.getToSignArtifact().equals(publishArtifact);
-                        }
-                    }));
+                    removeSignature(publishArtifact);
                 }
             });
         }
 
     }
 
-    private boolean addSignature(Signature signature) {
-        return signatures.add(signature);
+    /**
+     * Configures the task to sign every artifact of the given publication
+     */
+    void sign(PublicationInternal<?> publication) {
```

Comment:
❌ This should be public and take the public type and cast it internally.

Reply:
Done in 71838111765.